config.py

```diff
@@ -284,72 +393,86 @@ def __init__(self):
 
-            self.identifier = 'uk.org.marginal.%s' % appname.lower()
-
-            self.filename = join(getenv('XDG_CONFIG_HOME', expanduser('~/.config')), appname, '%s.ini' % appname)
+            self.filename = join(getenv('XDG_CONFIG_HOME', expanduser('~/.config')), appname, f'{appname}.ini')
             if not isdir(dirname(self.filename)):
                 makedirs(dirname(self.filename))
 
-            self.config = RawConfigParser(comment_prefixes = ('#',))
+            self.config = RawConfigParser(comment_prefixes=('#',))
             try:
                 with codecs.open(self.filename, 'r') as h:
                     self.config.read_file(h)
-            except:
+
+            except Exception as e:
+                logger.debug('config section not yet present', exc_info=e)
                 self.config.add_section(self.SECTION)
 
             if not self.get('outdir') or not isdir(self.get('outdir')):
                 self.set('outdir', expanduser('~'))
 
-        def get(self, key):
+        def get(self, key: str) -> Union[None, list, str]:
+            """Look up a string configuration value."""
             try:
                 val = self.config.get(self.SECTION, key)
-                if u'\n' in val:	# list
-                    # ConfigParser drops the last entry if blank, so we add a spurious ';' entry in set() and remove it here
+                if u'\n' in val:  # list
+                    # ConfigParser drops the last entry if blank,
+                    # so we add a spurious ';' entry in set() and remove it here
                     assert val.split('\n')[-1] == ';', val.split('\n')
                     return [self._unescape(x) for x in val.split(u'\n')[:-1]]
                 else:
                     return self._unescape(val)
-            except:
+
+            except Exception as e:
+                logger.debug('And the exception type is...', exc_info=e)
                 return None
 
-        def getint(self, key):
+        def getint(self, key: str) -> int:
+            """Look up an integer configuration value."""
             try:
                 return self.config.getint(self.SECTION, key)
-            except:
+
+            except Exception as e:
+                logger.debug('And the exception type is...', exc_info=e)
                 return 0
 
-        def set(self, key, val):
+        def set(self, key: str, val: Union[int, str, list]) -> None:
+            """Set value on the specified configuration key."""
             if isinstance(val, bool):
                 self.config.set(self.SECTION, key, val and '1' or '0')
+
             elif isinstance(val, str) or isinstance(val, numbers.Integral):
                 self.config.set(self.SECTION, key, self._escape(val))
-            elif hasattr(val, '__iter__'):	# iterable
+
+            elif isinstance(val, list):
                 self.config.set(self.SECTION, key, u'\n'.join([self._escape(x) for x in val] + [u';']))
```

Comment:
```suggestion
                self.config.set(self.SECTION, key, '\n'.join([self._escape(x) for x in val] + [u';']))
```

Reply:
And presumably `u';'` -> `';'` as well.

Reply:
Yeah, missed one :D

Reply:
Doing a search on u' and changing them all in this file....